optuna_dashboard/_app.py

```diff
@@ -449,6 +452,44 @@ def save_trial_note(study_id: int, trial_id: int) -> dict[str, Any]:
         response.status = 204  # No content
         return {}
 
+    @app.get("/csv/<study_id:int>")
+    def download_csv(study_id: int) -> BottleViewReturn:
+        # Create a CSV file
+        summary = get_study_summary(storage, study_id)
+        if summary is None:
+            response.status = 404  # Not found
+            return {"reason": f"study_id={study_id} is not found"}
+        trials = get_trials(storage, study_id)
+
+        param_names = sorted(set(chain.from_iterable([t.params.keys() for t in trials])))
+        user_attr_names = sorted(set(chain.from_iterable([t.user_attrs.keys() for t in trials])))
+
+        param_names_header = [f"Param {x}" for x in param_names]
+        user_attr_names_header = [f"UserAttribute {x}" for x in user_attr_names]
+        value_header = ["Value"]
+        if len(trials[0].values) > 1:
+            value_header = [f"Objective {x}" for x in range(len(trials[0].values))]
+        column_names = (
+            ["Number", "State"] + value_header + param_names_header + user_attr_names_header
+        )
+
+        buf = io.StringIO("")
+        writer = csv.writer(buf)
+        writer.writerow(column_names)
+        for frozen_trial in trials:
+            row = [frozen_trial.number, frozen_trial.state.name] + frozen_trial.values
+            row.extend([frozen_trial.params.get(name, None) for name in param_names])
+            row.extend([frozen_trial.user_attrs.get(name, None) for name in user_attr_names])
+            writer.writerow(row)
+
+        # Set response headers
+        response.headers["Content-Type"] = "text/csv; chatset=cp932"
+        response.headers["Content-Disposition"] = f"attachment; filename=trials_{study_id}.csv"
```

Comment:
How about using the `study_name` for the csv filename?

Reply:
I see. I will revise the code, with sanitizing the `study_name` :
```
        output_filename = re.sub(r'[\\/:*?"<>|]+', "", summary.study_name)
        response.headers["Content-Type"] = "text/csv; chatset=cp932"
        response.headers["Content-Disposition"] = f"attachment; filename={output_filename}.csv"
```